docs/scos/dev/front-end-development/migration-guide-upgrade-to-webpack-v5.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Migration guide - Upgrade to Webpack v5
+description: Use the guide to update versions of the Webpack and related modules.
+template: module-migration-guide-template
+---
+
+## Upgrading from version 4.* to version 5.*
+
+This document provides instructions for upgrading Webpack to version 5 in your Spryker project.
+
+## Overview
+
+The Webpack v5 was released 2 years ago.
+So we need to migrate as well to unblock future upgrades of other dependencies.
+
+*Estimated migration time: 1h*
+
+## 1) Update dependencies
+
+{% info_block infoBox "Note" %}
+
+To ensure that dependencies will install correctly, delete `node_modules` and `package-lock.json` before installation 
+using the next command `rm -rf node_modules && rm -rf package-lock.json`.
+
+{% endinfo_block %}
+
+1. Update the following dependencies to the new version:
+
+```json
+{
+    // update
+    "@types/webpack": "~5.28.0", // If using Merchant Portal
+    "clean-webpack-plugin": "~4.0.0",
+    "compression-webpack-plugin": "~10.0.0",
+    "copy-webpack-plugin": "~11.0.0",
+    "mini-css-extract-plugin": "~2.7.2",
+    "optimize-css-assets-webpack-plugin": "~6.0.1",
+    "terser-webpack-plugin": "~5.3.6",
+    "webpack": "~5.74.0",
+    "webpack-merge": "~5.8.0",
+    // add
+    "postcss": "~8.4.20"
+}
+```
+
+2. Run `npm install` to install dependencies and update `package-lock.json` file.
+
+## 2) Update configuration files
+
+**Do the following changes to update configuration files:**
+
+1. `frontend/configs/development-watch.js`:
+
+```js
+const merge = require('webpack-merge');
+// must be
+const { merge } = require('webpack-merge');
+```
+
+2. `frontend/configs/development.js`:
+
+- Rename the `output.jsonpFunction` property to the `chunkLoadingGlobal`.
+- Rename the `optimization.splitChunks.cacheGroups.vendors` property to the `defaultVendors`.
+
+3. `frontend/configs/production.js`:
+
+```js
+const merge = require('webpack-merge');
+// must be
+const { mergeWithCustomize, customizeObject } = require('webpack-merge');
+```
+
+```js
+const mergeWithStrategy = merge.smartStrategy({
+    plugins: 'prepend',
+});
+// must be
+const mergeWithStrategy = mergeWithCustomize({
+    customizeObject: customizeObject({
+        plugins: 'prepend',
+    })
+});
+```
+
+```js
+plugins: [
+    new CompressionPlugin({
+        filename: '[path].gz[query]',
+    }),
+
+    new CompressionPlugin({ // or `BrotliPlugin`
```

Comment:
```suggestion
    new CompressionPlugin({
```

Reply:
We still have BrotliPlugin plugin usage on demoshops, I assume customers have it as well

Reply:
This is not clear.
Also, I've checked 4 demo shops and they have: 
```
new BrotliPlugin({
                asset: '[path].br[query]',
                test: /\.js$|\.css$|\.svg$|\.html$/,
                threshold: 10240,
                minRatio: 0.8
            })
```
So let's have this in the example.